src/otx/core/data/entity/classification.py

```diff
@@ -144,3 +144,126 @@ def pin_memory(self) -> MultilabelClsBatchDataEntity:
+    :param label_to_idx: index of each label
+    :param empty_multiclass_head_indices: the index of head that doesn't include any label
+                                          due to the label removing
+
+    i.e.
+    Single-selection group information (Multiclass, Exclusive)
+    {
+        "Shape": ["Rigid", "Non-Rigid"],
+        "Rigid": ["Rectangle", "Triangle"],
+        "Non-Rigid": ["Circle"]
+    }
+
+    Multi-selection group information (Multilabel)
+    {
+        "Animal": ["Lion", "Panda"]
+    }
+
+    In the case above, HlabelInfo will be generated as below.
+    NOTE, If there was only one label in the multiclass group, it will be handeled as multilabel(Circle).
+
+        num_multiclass_heads: 2  (Shape, Rigid)
+        num_multilabel_classes: 3 (Circle, Lion, Panda)
+        head_to_logits_range: {'0': (0, 2), '1': (2, 4)} (Each multiclass head have 2 labels)
+        num_single_label_classes: 4 (Rigid, Non-Rigid, Rectangle, Triangle)
+        class_to_group_idx: {
+            'Non-Rigid': (0, 0), 'Rigid': (0, 1),
+            'Rectangle': (1, 0), 'Triangle': (1, 1),
+            'Circle': (2, 0), 'Lion': (2,1), 'Panda': (2,2)
+        } (head index, label index for each head)
+        all_groups: [['Non-Rigid', 'Rigid'], ['Rectangle', 'Triangle'], ['Circle'], ['Lion'], ['Panda']]
+        label_to_idx: {
+            'Rigid': 0, 'Rectangle': 1,
+            'Triangle': 2, 'Non-Rigid': 3, 'Circle': 4
+            'Lion': 5, 'Panda': 6
+        }
+        empty_multiclass_head_indices: []
+
+    All of the member variables should be considered for the Model API.
+    https://github.com/openvinotoolkit/training_extensions/blob/develop/src/otx/algorithms/classification/utils/cls_utils.py#L97
+    """
+
+    num_multiclass_heads: int
+    num_multilabel_classes: int
+    head_idx_to_logits_range: dict[str, tuple[int, int]]
+    num_single_label_classes: int
+    class_to_group_idx: dict[str, tuple[int, int]]
+    all_groups: list[list[str]]
+    label_to_idx: dict[str, int]
+    empty_multiclass_head_indices: list[int]
+
+
+@register_pytree_node
+@dataclass
+class HlabelClsDataEntity(OTXDataEntity):
+    """Data entity for H-label classification task.
+
+    :param labels: labels as integer indices
+    :param label_group: the group of the label
+    """
+
+    @property
+    def task(self) -> OTXTaskType:
+        """OTX Task type definition."""
+        return OTXTaskType.H_LABEL_CLS
+
+    labels: LongTensor
+    hlabel_info: HLabelInfo
+
+
+@dataclass
+class HlabelClsPredEntity(HlabelClsDataEntity, OTXPredEntity):
+    """Data entity to represent the H-label classification model output prediction."""
+
+
+@dataclass
+class HlabelClsBatchDataEntity(OTXBatchDataEntity[HlabelClsDataEntity]):
+    """Data entity for H-label classification task.
+
+    :param labels: A list of labels as integer indices
+    :param label_groups: A list of label group
+    """
+
+    labels: list[LongTensor]
+    hlabel_info: list[HLabelInfo]
```

Comment:
Is it should be (identical to all samples)?
```suggestion
    hlabel_info: HLabelInfo
```

Reply:
I removed hlabel_info field from HLabel releated entities since we could use the `HLabelMetaInfo`